reference_data/management/tests/update_all_reference_data_tests.py

```diff
@@ -16,33 +16,58 @@ def primate_ai_exception():
 def mgi_exception():
     raise Exception('MGI failed')
 
+SKIP_ARGS = [
+    '--skip-gencode', '--skip-dbnsfp-gene', '--skip-gene-constraint', '--skip-primate-ai', '--skip-mgi', '--skip-hpo',
+    '--skip-gene-cn-sensitivity',
+]
 
-@mock.patch('reference_data.management.commands.update_all_reference_data.logger')
-@mock.patch('reference_data.management.commands.update_all_reference_data.update_records')
-@mock.patch('reference_data.management.commands.update_all_reference_data.update_hpo')
-@mock.patch('reference_data.management.commands.update_all_reference_data.update_gencode')
-@mock.patch('reference_data.management.commands.update_all_reference_data.OmimReferenceDataHandler')
 class UpdateAllReferenceDataTest(TestCase):
     databases = '__all__'
     fixtures = ['users', 'reference_data']
 
-    @mock.patch('reference_data.management.commands.update_dbnsfp_gene.DbNSFPReferenceDataHandler')
-    @mock.patch('reference_data.management.commands.update_gene_constraint.GeneConstraintReferenceDataHandler')
-    @mock.patch('reference_data.management.commands.update_primate_ai.PrimateAIReferenceDataHandler')
-    @mock.patch('reference_data.management.commands.update_mgi.MGIReferenceDataHandler')
-    def test_update_all_reference_data_command(self, mock_mgi_handler, mock_primate_ai_handler, mock_gene_constraint_handler, mock_dbnsfp_gene_handler, mock_omim, mock_update_gencode, mock_update_hpo, mock_update_records, mock_logger):
+    def setUp(self):
+        patcher = mock.patch('reference_data.management.commands.update_dbnsfp_gene.DbNSFPReferenceDataHandler', lambda: 'dbnsfp_gene')
+        patcher.start()
+        self.addCleanup(patcher.stop)
```

Comment:
Is any reason to change the mocks from decorators to setUps?

Reply:
yes it allows them to be shared across all the test cleanly without passing tons of carefully ordered parameters that are hard to read and maintain and are often unused by most tests